airbyte-webapp/src/components/ui/CheckBox/CheckBox.tsx

```diff
@@ -29,7 +30,13 @@ export const CheckBox: React.FC<CheckBoxProps> = ({ indeterminate, checkboxSize
         className
       )}
     >
-      <input type="checkbox" aria-checked={indeterminate ? "mixed" : checked} {...inputProps} />
+      <input
+        type="checkbox"
+        aria-checked={indeterminate ? "mixed" : checked}
+        {...inputProps}
+        onClick={(e) => e.stopPropagation()}
```

Comment:
Looks like this is not working as expected, or does this PR need to be rebased/merge master? Right now clicking a checkbox is also triggering e.g. the stream panel to open.

https://user-images.githubusercontent.com/7550957/215805187-71b7e032-af64-4817-8311-a04a2b47c96a.mov

Reply:
Fun fact: `e.stopPropagation()` works here for keyboard nav but not mouse... and `event.nativeElement.stopImmediatePropagation()` works here for mouse but not keyboard.  Neat.
(Also TIL [nativeElement.stopImmediatePropagation()](https://gist.github.com/ggregoire/ce7bc946212920c0a6bad8125567d001)).  To be continued....

Reply:
I'm in general a bit hesistant to put the stopPropagation logic into the core component itself. Us nesting interactive inputs (against best practice), I think should not cought on the base component levels (and thus prevent with standard browser behavior of inputs), but I'd rather suggest that we actually let the component bubble the event freely, and the consuming component that has problems caused by it (because it nested it inside another interactive element), must make sure to handle this properly (e.g. by checking in it's `onClick` event that the actual `ev.target` wasn't actually another interactive element).

Happy though to put that up for discussion and get more people's feeling, if we should or shouldn't handle event propagation stops inside core UI components.

Reply:
That makes sense... I wasn't able to get it working this way anyways, even after pairing with @ambirdsall on it for quite some time.

Reply:
Leaving this open for posterity's sake, but all bits related to checkboxes have been removed from this PR.